Algorithms/Dynamic-Programming/modified_knapsack.cpp

```diff
@@ -0,0 +1,135 @@
+           				if((val[i-1] + K[i-1][w-wt[i-1]].maxval)>K[i-1][w].maxval){
+           					// checking if max items limit is exceeded or not
+           					if((1+K[i-1][w-wt[i-1]].maxitems)<=mit){
+           						K[i][w].maxval = val[i-1] + K[i-1][w-wt[i-1]].maxval;
+           						K[i][w].maxitems = 1+K[i-1][w-wt[i-1]].maxitems;
+           					}
+           					else{
+           						// selecting current item if it has max value
+           						if(val[i-1]>=K[i-1][w].maxval && val[i-1]>=K[i-1][w-wt[i-1]].maxval){
+           							K[i][w].maxval=val[i-1];
+           							K[i][w].maxitems=1;
+           						}
+           						else{
+           							if(K[i-1][w].maxval>K[i-1][w-wt[i-1]].maxval){
+           								K[i][w].maxval=K[i-1][w].maxval;
+           								K[i][w].maxitems = K[i-1][w].maxitems;
+           							}
+           							else if(K[i-1][w].maxval<K[i-1][w-wt[i-1]].maxval){
+           								K[i][w].maxval=K[i-1][w-wt[i-1]].maxval;
+           								K[i][w].maxitems = K[i-1][w-wt[i-1]].maxitems;
+           							}
+           							else{
+           								//selecting item which has minimum items taken already
+           								if(K[i-1][w].maxitems<=K[i-1][w-wt[i-1]].maxitems){
+           									K[i][w].maxval=K[i-1][w].maxval;
+           									K[i][w].maxitems = K[i-1][w].maxitems;
+           								}
+           								else{
+           									K[i][w].maxval=K[i-1][w-wt[i-1]].maxval;
+           									K[i][w].maxitems = K[i-1][w-wt[i-1]].maxitems;
+           								}
+           							}
+           						}
+           					}
+           				}
+           				//not taking current item in knapsack
+           				else if((val[i-1] + K[i-1][w-wt[i-1]].maxval) < K[i-1][w].maxval){
+            				K[i][w].maxval = K[i-1][w].maxval;
+            				K[i][w].maxitems = K[i-1][w].maxitems;
+           				}
+           				else{
+           					if(K[i-1][w-wt[i-1]].maxitems<=K[i-1][w].maxitems){ 
+           						if((1+K[i-1][w-wt[i-1]].maxitems)<=mit){
+           							K[i][w].maxval = val[i-1] + K[i-1][w-wt[i-1]].maxval;
+           							K[i][w].maxitems = 1+K[i-1][w-wt[i-1]].maxitems;
+           						}
+           						else{
+           							K[i][w].maxval = K[i-1][w].maxval;
+           							K[i][w].maxitems = K[i-1][w].maxitems;	
+           						}
+           					}
+           					else{
+           						K[i][w].maxval = K[i-1][w].maxval;
+           						K[i][w].maxitems = K[i-1][w].maxitems;
+           					}
+           				}
+			}
+			// if weight of current item exceeds knapsack weight in consideration, we simply take
+			// the best case for 1 the items that have been checked already for the same knapsack capacity
+			else{
+            			K[i][w].maxval = K[i-1][w].maxval;
+            			K[i][w].maxitems = K[i-1][w].maxitems;
+			}
+		}
+	}
+	cout<< "Maximum weight possible : " << K[n][W].maxval << endl;
```

Comment:
Do you mean maximum profit/value possible?

Reply:
Yes, mentioned weight by mistake. I'll make that change. Is there any other change that I need to make ?